### PR TITLE
SCB-96 added interceptor for rest template

### PR DESCRIPTION
**Pull request by eric-lee-ltk**
Signed-off-by: Eric Lee <dagang.li@huawei.com>

### CODE REVIEW

File: omega/omega-transport/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (seanyinx) [inline omega/omega-transport/pom.xml:1]: please add apache license headers

File: omega/omega-transport/src/main/java/io/servicecomb/saga/omega/transport/TransactionClientHttpRequestInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import io.servicecomb.saga.core.IdGenerator;
+
+public class TransactionClientHttpRequestInterceptor implements ClientHttpRequestInterceptor {
+
+  public static String TRANSACTION_ID_KEY = "X-Transaction-Id";
+
+  private IdGenerator<String> randomIdGenerator = new UniqueIdGenerator();
```
Comment (seanyinx) [inline omega/omega-transport/src/main/java/io/servicecomb/saga/omega/transport/TransactionClientHttpRequestInterceptor.java]: please make it final. we may provide other implementation of `IdGenerator`. how about passing it in from constructor?

File: omega/omega-transport/src/test/java/io/servicecomb/saga/omega/transport/UniqueIdGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import org.junit.Test;
+
+public class UniqueIdGeneratorTest {
+
+  private UniqueIdGenerator idGenerator = new UniqueIdGenerator();
+
+  @Test
+  public void nextIdIsUnique() {
+    Set<String> ids = new HashSet<>();
+    for (int i = 0; i < 10; i++) {
```
Comment (seanyinx) [inline omega/omega-transport/src/test/java/io/servicecomb/saga/omega/transport/UniqueIdGeneratorTest.java]: does this test guarantee the id generator will generate unique ids in multithreaded scenario?

File: omega/pom.xml [hunk 4]
```diff
@@ -32,6 +32,7 @@
     <module>omega-transaction</module>
     <module>omega-context</module>
     <module>omega-spring-tx</module>
+    <module>omega-transport</module>
```
Comment (seanyinx) [inline omega/pom.xml:35]: this implementation is for spring rest template only. we may need to provide other implementations, such as thrift, grpc. it's better to make a transport hierarchy like:
```
omega-transport
|--omega-transport-resttemplate
|--others-in-the-future
```

File: omega/omega-transport/pom.xml (hunk 1 above)
Comment (eric-lee-ltk) [reply omega/omega-transport/pom.xml:1]: done

File: omega/pom.xml (hunk 4 above)
Comment (eric-lee-ltk) [reply omega/pom.xml:35]: done

**Pull request merged**